src/table/components/head/HeadCellMenu.tsx

```diff
@@ -1,110 +1,145 @@
+                  },
+                  icon: <SearchIcon />,
+                  isDisabled: false,
+                },
+              ],
+            },
+          ]
+        : []),
+    ],
+    [translator, isInteractionEnabled, isCurrentColumnActive, sortDirection, isDimension]
+  );
 
   return (
     <StyledCellMenu headerStyle={headerStyle}>
       <StyledMenuIconButton
         ref={anchorRef}
         tabIndex={-1}
         id="sn-table-head-menu-button"
-        aria-controls={open ? 'sn-table-head-menu' : undefined}
-        aria-expanded={open ? 'true' : undefined}
+        aria-controls={openPrimaryDropdown ? 'sn-table-head-menu' : undefined}
+        aria-expanded={openPrimaryDropdown ? 'true' : undefined}
         aria-haspopup="true"
-        onClick={() => setOpen(!open)}
+        onClick={() => setOpenPrimaryDropdown(!openPrimaryDropdown)}
       >
         <MoreHoriz />
       </StyledMenuIconButton>
       <Popper
-        modifiers={[
-          {
-            name: 'offset',
-            options: {
-              offset: [0, 9],
-            },
-          },
-        ]}
-        open={open}
+        modifiers={[{ name: 'offset', options: { offset: [0, 9] } }]}
+        open={openPrimaryDropdown}
         anchorEl={anchorRef.current}
         role={undefined}
         placement="bottom-start"
         transition
         disablePortal
       >
-        {({ TransitionProps, placement }) => (
-          <Grow
-            {...TransitionProps}
-            style={{
-              transformOrigin: placement === 'bottom-start' ? 'left top' : 'left bottom',
-            }}
-          >
-            <Paper sx={{ boxShadow: 15 }}>
-              <ClickAwayListener onClickAway={() => setOpen(false)}>
+        {({ TransitionProps }) => (
+          <Grow {...TransitionProps} style={{ transformOrigin: 'left top' }}>
+            <PrimaryDropdownPaper sx={{ boxShadow: 15 }}>
+              <ClickAwayListener onClickAway={() => setOpenPrimaryDropdown(false)}>
                 <MenuList
-                  autoFocusItem={open}
+                  autoFocusItem={openPrimaryDropdown}
                   className="sn-table-head-menu"
                   aria-labelledby="sn-table-head-menu-button"
                 >
-                  <MenuItem itemTitle={translator.get('SNTable.MenuItem.SortAscending')} newSortDirection="A">
-                    <ArrowUpwardIcon />
-                  </MenuItem>
-
-                  <MenuItem itemTitle={translator.get('SNTable.MenuItem.SortDescending')} newSortDirection="D">
-                    <ArrowDownwardIcon />
-                  </MenuItem>
+                  {menuItems.map((menuGroup, i) => (
+                    <MenuGroup key={menuGroup.id} {...menuGroup} shouldShowDevider={i !== menuItems.length - 1} />
+                  ))}
                 </MenuList>
               </ClickAwayListener>
+            </PrimaryDropdownPaper>
+          </Grow>
+        )}
+      </Popper>
+
+      <Popper
```

Comment:
maybe move this to the listbox wrapper? Then you wouldn't have to pass the ref like you do here.

Reply:
In my opinion no, for a few reasons: 
1. That would make us to pass so many props into ListBox Wrapper which might not be necessary at all
2. if we do so, List box wrapper has to handle the Popper stuff/logic as well
3. This is a common pattern in react which currently reduces a lot's of overhead in implementation phase (the most important one was the replacement of the previous hook and separating the concern of listbox from it's parent dropdown) by only isolating the embed/mount/unmount logic of only listbox

Reply:
can't argue with that, let's keep it here